Src/FluentAssertions/Specialized/ActionAssertions.cs

```diff
@@ -166,8 +166,8 @@ public void NotThrowAfter(TimeSpan waitTime, TimeSpan pollInterval, string becau
                     return;
                 }
 
-                Thread.Sleep(pollInterval);
                 invocationEndTime = watch.Elapsed;
+                Thread.Sleep(pollInterval);
```

Comment:
🤔I don't think these `Thread.Sleep`s should have been moved.
When checking the loop condition the second time, `invocationEndTime` would be e.g. `x` while `Subject` was first invoked `x + pollInterval` ago. (If that makes any sense)

Reply:
I don't h ave any clue what I was doing. But I can't think of any theoretical reason why it would fail on AppVeyor 😢

Reply:
I think it's failing on AppVeyor and not locally because presumably the build agent is less powerful than your machine. 
I think this a similar issue to other methods related to timing failing occasionally. We're relying on something being executed either within a precise amount of time, or to be executed after waiting a precise amount of time. I've found that these tests can be unreliable, especially when running other things on the same machine, or running it on a slower machine. 

I'm also looking into possible fixes for this. I'll let you know if I have anything.